src/solvers/dgsem_tree/dg_2d_subcell_limiters.jl

```diff
@@ -14,20 +14,42 @@ function create_cache(mesh::TreeMesh{2}, equations,
     A3dp1_x = Array{uEltype, 3}
     A3dp1_y = Array{uEltype, 3}
     A3d = Array{uEltype, 3}
-
-    fhat1_threaded = A3dp1_x[A3dp1_x(undef, nvariables(equations), nnodes(dg) + 1,
-                                     nnodes(dg)) for _ in 1:Threads.nthreads()]
-    fhat2_threaded = A3dp1_y[A3dp1_y(undef, nvariables(equations), nnodes(dg),
-                                     nnodes(dg) + 1) for _ in 1:Threads.nthreads()]
+    A4d = Array{uEltype, 4}
+
+    fhat1_L_threaded = A3dp1_x[A3dp1_x(undef, nvariables(equations), nnodes(dg) + 1,
+                                       nnodes(dg)) for _ in 1:Threads.nthreads()]
+    fhat2_L_threaded = A3dp1_y[A3dp1_y(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg) + 1) for _ in 1:Threads.nthreads()]
+    fhat1_R_threaded = A3dp1_x[A3dp1_x(undef, nvariables(equations), nnodes(dg) + 1,
+                                       nnodes(dg)) for _ in 1:Threads.nthreads()]
+    fhat2_R_threaded = A3dp1_y[A3dp1_y(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg) + 1) for _ in 1:Threads.nthreads()]
     flux_temp_threaded = A3d[A3d(undef, nvariables(equations), nnodes(dg), nnodes(dg))
                              for _ in 1:Threads.nthreads()]
+    flux_temp_nonconservative_threaded = A4d[A4d(undef, nvariables(equations),
+                                                 nnoncons(equations),
+                                                 nnodes(dg), nnodes(dg))
+                                             for _ in 1:Threads.nthreads()]
+    fhat_temp_threaded = A3d[A3d(undef, nvariables(equations), nnodes(dg),
+                                 nnodes(dg))
+                             for _ in 1:Threads.nthreads()]
+    fhat_nonconservative_temp_threaded = A4d[A4d(undef, nvariables(equations),
+                                                 nnoncons(equations),
+                                                 nnodes(dg), nnodes(dg))
+                                             for _ in 1:Threads.nthreads()]
```

Comment:
I think we should use the same variable name structure for all variables. So, probably rename `flux_temp_nonconservative_threaded` to `flux_nonconservative_temp_threaded`.

Reply:
Thanks! I renamed the variables in 20abe7b